src/components/gitOps/UserGitRepConfiguration.tsx

```diff
@@ -0,0 +1,150 @@
+import { toast } from 'react-toastify'
+import { URLS } from '../../config'
+import { ReloadNoGitOpsRepoConfiguredModal } from '../workflowEditor/NoGitOpsRepoConfiguredWarning'
+
+export default function UserGitRepConfiguration(props: UserGitRepoConfigurationProps) {
+    const [gitOpsRepoURL, setGitOpsRepoURL] = useState('')
+    const [selectedRepoType, setSelectedRepoType] = useState(repoType.DEFAULT)
+    const [isEditable, setIsEditable] = useState(false)
+    const [showReloadModal, setShowReloadModal] = useState(false)
+    const [loading, setLoading] = useState(false)
+    const history = useHistory()
+
+    useEffect(() => {
+        setLoading(true)
+        getGitOpsRepoConfig(props.appId)
+            .then((response) => {
+                if (response.result) {
+                    setGitOpsRepoURL(response.result.gitRepoURL)
+                    setIsEditable(response.result.isEditable)
+                }
+            })
+            .catch((err) => {
+                showError(err)
+            })
+            .finally(() => {
+                setLoading(false)
+            })
+    }, [])
+
+    const renderSavedGitOpsRepoState = (repoURL) => (
+        <>
+            {loading ? (
+                <div className="w-100 h-100">
+                    <Progressing pageLoader />
+                </div>
+            ) : (
+                <div className="pt-16 pl-20">
+                    <div>
+                        <div className="fw-4 fs-13 fcn-9">
+                            Application Deployemnt states are saved as manifest in a Git repository. ArgoCD uses these
+                            manifests to sync with your live Kubernetes cluster.
+                        </div>
+                        <div className="fs-13 fw-4 flexbox-col mt-16 mb-16">
+                            <div className="">Configurations for this application will be committed to:</div>
+                            <a
+                                className="dc__ff-monospace dc__link "
+                                href={repoURL}
+                                target="_blank"
+                                rel="noreferrer noopener"
+                            >
+                                {repoURL}
+                            </a>
+                        </div>
+                    </div>
+                    {renderInfoColorBar()}
+                </div>
+            )}
+        </>
+    )
+
+    const renderInfoColorBar = () => {
+        return (
+            <InfoColourBar
+                message="GitOps repository for this application is immutable once saved."
+                classname="warn"
+                Icon={Warn}
+                iconClass="warning-icon"
+            />
+        )
+    }
+
+    const closePopup = () => {
+        setShowReloadModal(false)
+    }
+
+    const reload = () => {
+        history.push(`/app/${props.appId}/edit/${URLS.APP_WORKFLOW_CONFIG}`)
+        window.location.reload()
```

Comment:
Why are we doing page reload here?

Reply:
As per the requirement if a case of stale data occurs than we have to reload the page.